server/src/session.ts

```diff
@@ -125,6 +125,11 @@ export class Session {
       return;
     }
     project.enableLanguageService();
+    // When the language service got disabled, the program was discarded via
+    // languageService.cleanupSemanticCache(). However, the program is not
+    // recreated when the language service is re-enabled. We manually mark the
+    // project as dirty to force update the graph.
+    project.markAsDirty();
```

Comment:
I don't like all these "gotcha"s we keep running into with internal TS apis that are not well documented. Maybe another reason to migrate away from relying on the plugin model.

Reply:
Do you mean we rewrite the server ourselves? We need some kind of integration with TypeScript, but so far it is not clear to me where the boundary should be. Even if we don't use the tsserver plugin model, we still rely on tsserver for project management and document tracking. Is there an ideal architecture you have in mind?

Reply:
For example to avoid this problem, if we avoid the plugin model, we can have a project and keep the tsLs for the project always enabled. Then we can enable/disable the angular language service as needed without having to go through the project, and the project will still update the tsLs language service (and hence the program) as needed. A potential concern is that the project may then be updating the tsLs language service more often than necessary, but I think in practice this will not be an issue.

Reply:
Ah I see what you mean - you're suggesting we maintain two distinct language service instances - tsserver maintains its default TypeScript language service, and we keep our own Angular language service. As it is right now, there is really not a way for us to distinguish one from another. From the project point of view, there is only one combined LS, and therefore one combined state. Yeah, I can see the two-fold benefits of separating them (distinct state + our own LS interface). Let's discuss this more when we meet in person 😄